src/database/queries/modules.js

```diff
@@ -0,0 +1,75 @@
+const Modules = require('@database/models/index').Module
+const { Op } = require('sequelize')
+
+module.exports = class UserRoleModulesData {
+	static async createModules(data) {
+		try {
+			return await Modules.create(data, { returning: true })
+		} catch (error) {
+			throw error
+		}
+	}
+
+	static async findModulesById(id) {
+		try {
+			return await Modules.findByPk(id)
+		} catch (error) {
+			throw error
+		}
+	}
+
+	static async findAllModules(page, limit, search) {
+		try {
+			const offset = (page - 1) * limit
+
+			const whereCondition = {
+				code: { [Op.iLike]: `%${search}%` },
+			}
+
+			const options = {
+				where: whereCondition,
+				offset,
+				limit,
+				attributes: ['id', 'code', 'status'],
+			}
+
+			const permissions = await Modules.findAndCountAll(options)
+			return permissions
+		} catch (error) {
+			throw error
+		}
+	}
+
+	static async updateModulesById(id, updatedata) {
```

Comment:
@sumanvpacewisdom make this function as generic, so that where condition can be changes based on the requirement